utils/tqueue/tqueue.h

```diff
@@ -0,0 +1,90 @@
+#include <queue>
+#include <mutex>
+
+
+template<typename T>
+class tqueue {
+    std::queue<T> internal_queue;
+    std::mutex queue_write_mutex;
+
+    public: 
+
+    tqueue() : internal_queue(), queue_write_mutex() {}
+
+
+    T& front(){
+        return this->internal_queue.front();
```

Comment:
Reader functions need mutexes too i would say, what if you call front while another thread deletes the item

Reply:
Yes. What do you think about empty and size, should the mutex be set there too? I think we should just do all the functions.

Reply:
Yes. They also need it.